src/cargo/util/network.rs

```diff
@@ -1,4 +1,30 @@
-use util::{CargoResult, Config, errors};
+use util::Config;
+use util::errors::{CargoError, CargoErrorKind, CargoResult};
+
+use git2;
+
+fn maybe_spurious(err: &CargoError) -> bool {
+    match err.kind() {
```

Comment:
Could this be modified to iterate over the error to see if anything along the way is spurious? In theory if there's any spurious error in the chain then this operation should be retried